Fix Partition Manager dependency

ArekBalysNordic opened the pull request:
- Removed obsolete CMake files that are not used anywhere.
- Fixed dependency on factory data partition when the Partition Manager is not enabled.
- Fixed kSettingsBlockEnd when Partition Manager is not enabled.

### Testing

In NCS

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/nrfconnect/sdk-connectedhomeip?pullRequest=682) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/nrfconnect/sdk-connectedhomeip?pullRequest=682) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: ArekBalysNordic<br/>:x: nordicjm<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/nrfconnect/sdk-connectedhomeip?pullRequest=682) it.</sub>

ArekBalysNordic committed:
[nrf toup] Migrate to non-PM builds

- There should not be dependency to the Partition Manager if it is
not enabled.

- Fix wrongly referring to partitions as alises

- To properly resolve `@{ZEPHYR_NRF_MODULE_DIR}` we need to use
string(CONFIGURE ...) on it.

- Add Kconfig to set the proper CHIP_APP_ZAP_DIR if the
custom location of the zap-generated files is used.

- Fix Factory Data to put files in the correct place and run
with the correct configuration

Signed-off-by: Arkadiusz Balys <arkadiusz.balys@nordicsemi.no>
Co-authored-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>
